### DISCUSSION TITLE
RedirectResponse to service hosted on a container timing out

### DISCUSSION THREAD

**Discussion by xiki808** [fastapi/fastapi#6513] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get("/container/somepath")
async def container():
   return RedirectResponse("https://10.0.0.4:10401/somepath")
```

### Description

I'm having an issue redirecting the request using the containers' IP, the request it redirected but the response times out. I'm not sure if this can be fixed from my FastAPI project or from my server setup, but when testing with curl on the server, I get the right response data.

I am running a number of containers on a server, behind a reverse proxy which is running in another container. My FastAPI project is stored on the host, but the requests go through the reverse proxy container, where the FastAPI socket is mapped and consumed by nginx. 

In the API, when I do a RedirectResponse using the host IP I get a response, but I would like to use the containers' IP instead. The containers' IP are not visible on the LAN (only from host or between them), but I can access them through port forwarding. If I manage to redirect using the internal IPs, I would remove port forwarding, and access these services through my FastAPI project, 

The problem is that even though I can curl an endpoint from a container on the host or on the reverse proxy container using the internal IPs, the redirection is not working through FastAPI. Is there something I'm missing? Not sure what else I can do to debug this.

### Operating System

Linux

### FastAPI Version

0.73.0

### Python Version

3.9

**Comment by xiki808** [comment]
Silly me.. testing things on the browser I got better error messages. The issue is that I'm redirecting HTTPS to HTTP. I cannot serve the container service over HTTPS directly, that's why I have the reverse proxy ( which listens to the host IP, available on the LAN ). 

Is there a way how to force HTTPS to HTTP?

**Comment by jgould22** [comment]
Your explanation is a little confusing. 

I would not be redirecting things to ips within dockers internal networks ( I would generally use dockers internal dns) and you should probably give each of your services a hostname (service1.whateverthishostiscalled.com, service2...) and redirect to those.

The browsers will also probably throw a fit if you try and swap https for http but I have not tried it recently.

Perhaps you can add a diagram so the end goal is a little easier to grok.

**Comment by xiki808** [comment]
Thanks for following up @jgould22 and excuse my confusing question. As you suggested, I'm going  to include a diagram, with hopes to be more understandable.

![Network drawio (3)](https://user-images.githubusercontent.com/8024321/151969331-31801c4f-86ab-4724-a6ba-3f2e6882559d.png)


From the image, the blue and red boxes are the containers running on the server, each running a specific service. The Reverse Proxy container holds all the port forwarding mapping. So for a request, the port is used to identify which service nginx will redirect to, adding SSL while doing so. The Fast API project is a new addition, which I'm hoping to add as a tokenized gateway layer (and more), between the reverse proxy and the services.

1. A request is sent example https://10.0.0.4:1234/api/get
2. In this example, port 1234 is meant to reach the Fast API project. The request is first forwarded to the Reverse Proxy container. 
3. Here nginx is setup to add SSL and then redirects it to http://10.0.0.4:1000, which is that of the Fast API project
4. The request arrives at Fast API project and is redirected to a service running on a container with:

```
@app.get("/service1/{path:path}")
async def service1(request: Request, path: str):
    return RedirectResponse(f"http:/10.0.2.2:8888/{path}")
```

The request is reaching Fast API but the response is never received, it is timing out. Most probably the issue is with my redirection from HTTPS-to-HTTP, but I don't need HTTPS for the service hosted in a container - SSL is already provided before the request reaches FAST API project. Is there a way how to force HTTPS-to-HTTP redirection? I will also be interested to do redirections of websockets.

**Comment by xiki808** [comment, marked as answer]
Well I did this and it works fine

```
@app.get("/service1/{path:path}")
async def service1(request: Request, path: str):
    return json.loads(requests.get(f"http://10.0.2.2:5000/{path}").content.decode('utf-8'))
```

**Comment by jgould22** [comment]
That is good.

One note, you will want to use a async http client. Requests is sync so your process will stop at that call and not work on other things.

**Comment by xiki808** [comment]
Appreciate that tip, thank you!
